Hpcc 16083 Provide ESP Binding information from WsESDLConfig

rpastrana opened the pull request.

rpastrana committed:
HPCC-16083 Interface enhancements to help UI development

Signed-off-by: rpastrana <rodrigo.pastrana@lexisnexis.com>

rpastrana committed:
HPCC-16083 Add new List ESP bindings of type desdl method

Signed-off-by: rpastrana <rodrigo.pastrana@lexisnexis.com>

rpastrana commented:
@wangkx please review

HPCCSmoketest commented:
Automated Smoketest 
Sha: f7a718c996095637945830719c43004a1b76145c
Build: success
Install hpccsystems-platform-community_6.1.0-trunk0.el6.x86_64.rpm
HPCC Start: OK
Unittest result: total:82 passed:81 failed:1

HPCC Stop: OK
HPCC Uninstall: OK

wangkx commented on esp/scm/ws_esdlconfig.ecm:
```diff
@@ -47,6 +47,17 @@ ESPresponse [exceptions_inline] PublishESDLDefinitionResponse
 
+ESPStruct MethodConfig
+{
+	String Name;
```
indent 8?

wangkx commented on esp/scm/ws_esdlconfig.ecm:
```diff
@@ -144,7 +146,7 @@ ESPresponse [exceptions_inline] GetESDLBindingResponse
     string BindingName;
     string EspPort;
-    [depr_var("1.1")] String ConfigXML;
+    String ConfigXML;
```
What if 1.1?

rpastrana replied:
I should have never deprecated... that change was never released so we'll always return ConfigXML

wangkx commented on esp/services/ws_esdlconfig/ws_esdlconfigservice.cpp:
```diff
@@ -615,7 +616,28 @@ bool CWsESDLConfigEx::onPublishESDLBinding(IEspContext &context, IEspPublishESDL
+                }
+                methodsxml.append("/>");
+            }
+            methodsxml.append("</Methods>");
```
Do you need to check if (methods.ordinality() > 0) here?

wangkx commented on esp/services/ws_esdlconfig/ws_esdlconfigservice.cpp:
```diff
@@ -615,7 +616,28 @@ bool CWsESDLConfigEx::onPublishESDLBinding(IEspContext &context, IEspPublishESDL
+                methodsxml.append("/>");
+            }
+            methodsxml.append("</Methods>");
+            methodstree.setown(createPTreeFromXMLString(methodsxml.str()));
```
And here?

wangkx commented on esp/services/ws_esdlconfig/ws_esdlconfigservice.cpp:
```diff
@@ -823,6 +843,21 @@ bool CWsESDLConfigEx::onConfigureESDLBindingMethod(IEspContext &context, IEspCon
+        {
+            StringBuffer methodsxml("<Methods>");
+
+            IConstMethodConfig& method = req.getMethodStructure();
```
What if no MethodStructure is set in req?

wangkx commented on esp/scm/ws_esdlconfig.ecm:
```diff
@@ -85,6 +86,60 @@ ESPresponse [exceptions_inline] PublishESDLBindingResponse
 
+ESPrequest ListDESDLEspBindingsReq
+{
+	boolean IncludeESDLBindingInfo;
```
indent 8?

wangkx commented on esp/scm/ws_esdlconfig.ecm:
```diff
@@ -85,6 +86,60 @@ ESPresponse [exceptions_inline] PublishESDLBindingResponse
+
+ESPStruct ESPServerEx
+{
+	string Name;
```
indent

wangkx commented on esp/services/ws_esdlconfig/ws_esdlconfigservice.cpp:
```diff
@@ -1347,6 +1348,116 @@ bool CWsESDLConfigEx::onListESDLDefinitions(IEspContext &context, IEspListESDLDe
 
+bool CWsESDLConfigEx::onListDESDLEspBindings(IEspContext &context, IEspListDESDLEspBindingsReq&req, IEspListDESDLEspBindingsResp &resp)
+{
+	bool includeESDLBindings = req.getIncludeESDLBindingInfo();
```
What if not set in req?

wangkx commented on esp/services/ws_esdlconfig/ws_esdlconfigservice.cpp:
```diff
@@ -1347,6 +1348,116 @@ bool CWsESDLConfigEx::onListESDLDefinitions(IEspContext &context, IEspListESDLDe
 }
 
+bool CWsESDLConfigEx::onListDESDLEspBindings(IEspContext &context, IEspListDESDLEspBindingsReq&req, IEspListDESDLEspBindingsResp &resp)
+{
```
Wrong indent inside this function

wangkx commented on esp/services/ws_esdlconfig/ws_esdlconfigservice.cpp:
```diff
@@ -1347,6 +1348,116 @@ bool CWsESDLConfigEx::onListESDLDefinitions(IEspContext &context, IEspListESDLDe
+									methodconfig->setName(attributes->queryValue());
+								}
+								else
+								{
```
Move line 1411 here?

wangkx commented on esp/services/ws_esdlconfig/ws_esdlconfigservice.cpp:
```diff
@@ -1347,6 +1348,116 @@ bool CWsESDLConfigEx::onListESDLDefinitions(IEspContext &context, IEspListESDLDe
+					if (esdlbindingtree)
+					{
+						Owned<IPropertyTree> def;
+						def.setown(esdlbindingtree->queryPropTree("Definition[1]"));
```
Do you need to check the 'def' before using it?

wangkx commented on esp/services/ws_esdlconfig/ws_esdlconfigservice.cpp:
```diff
@@ -1347,6 +1348,116 @@ bool CWsESDLConfigEx::onListESDLDefinitions(IEspContext &context, IEspListESDLDe
+								}
+								else
+								{
+									iespattribute->setName(attributes->queryName()+1);
```
You may use 'attname' here.

wangkx commented:
@rpastrana looks good. A few questions.

rpastrana committed:
HPCC-16083 Code review changes

- Addresses formating issues
- Addresses nil entries in XML requests
- Fixes versioning
- Makes SDS_LOCK_TIMEOUT unique name: SDS_LOCK_TIMEOUT_DESDL


Signed-off-by: rpastrana <rodrigo.pastrana@lexisnexis.com>

rpastrana commented:
@wangkx please review changes based on code review.

HPCCSmoketest commented:
Automated Smoketest 
Sha: 70c5e24ec6a69202ef7545be6eae99ee14c93f6b
Build: success
Install hpccsystems-platform-community_6.1.0-trunk0.el6.x86_64.rpm
HPCC Start: OK
Unittest result: total:82 passed:81 failed:1

HPCC Stop: OK
HPCC Uninstall: OK

wangkx commented on esp/services/ws_esdlconfig/ws_esdlconfigservice.cpp:
```diff
@@ -1350,117 +1366,116 @@ bool CWsESDLConfigEx::onListESDLDefinitions(IEspContext &context, IEspListESDLDe
+                        Owned<IPropertyTree> def;
+                        def.setown(esdlbindingtree->queryPropTree("Definition[1]"));
+
+                        StringBuffer defid = def->queryProp("@id");
```
Is it possible that def is NULL?

wangkx commented on esp/services/ws_esdlconfig/ws_esdlconfigservice.cpp:
```diff
@@ -1350,117 +1366,116 @@ bool CWsESDLConfigEx::onListESDLDefinitions(IEspContext &context, IEspListESDLDe
+
+                        msg.appendf("\nFetched ESDL Biding Configuration for %d methods.", iesmethods.length());
+
+						StringBuffer definition;
```
Wrong indent

wangkx commented:
@rpastrana 2 comments

rpastrana committed:
HPCC-16083 Addressed issues from 2nd code review

Signed-off-by: rpastrana <rodrigo.pastrana@lexisnexis.com>

rpastrana commented:
@wangkx one more review please

HPCCSmoketest commented:
Automated Smoketest 
Sha: 96d461b76a1961fc03adc5e7fe991523be0bf28e
Build: success
Install hpccsystems-platform-community_6.1.0-trunk0.el6.x86_64.rpm
HPCC Start: OK
Unittest result: total:82 passed:81 failed:1

HPCC Stop: OK
HPCC Uninstall: OK

wangkx commented:
@rpastrana looks fine.

rpastrana commented:
@richardkchapman please merge.
